exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -194,10 +145,13 @@ func (b *Bitfinex) wsHandleData(respRaw []byte) error {
 		eventType, hasEventType := d[1].(string)
 
 		if chanID != 0 {
-			if c, ok := b.WebsocketSubdChannels[chanID]; ok {
-				return b.handleWSChannelUpdate(c, chanID, eventType, d)
+			if c := b.Websocket.GetSubscription(chanID); c != nil {
+				return b.handleWSChannelUpdate(c, eventType, d)
 			}
-			return fmt.Errorf("unable to locate chanID: %d", chanID)
+			// We didn't have a mapping for this chanID; This probably means we have unsubscribed OR
+			// received our first message before processing the sub chanID
+			// In either case it's okay. No point in erroring because there's nothing we can do about it, and it happens often
```

Comment:
I think something like `if b.verbose { log.Warnf(log.Websocket, "dropped ws update %s", d) }` would be good. One has to opt in seeing if its worth optimising to avoid what gets dropped. eg a raw orderbook snapshot gets dropped and then we fail orderbook processing and requiring a whole unsub/resub cycle to resolve

Reply:
Yeah, I think that would make sense too.
I've used:
```
log.Warnf(log.ExchangeSys, "%s %s; dropped WS message: %s", b.Name, stream.ErrSubscriptionNotFound, respRaw)
```
This brings forward a export of ErrSubscriptionNotFound already in PR #1358 
Fixed 70ce75a7